File: next-dashboard-ui/src/lib/exam.ts
import { Subject } from "./subject";
import { Class } from "./class";
import { Teacher } from "./teacher";

export interface Exam {
  examID: number;
  examName?: string;
  examDate?: string;

  subjectID?: number;
  subject?: Subject;

  classID?: number;
  class?: Class;

  teacherID?: number;
  teacher?: Teacher;
}

File: backend-asp-school/backend-asp-school/Models/Exam.cs
using System.Security.Claims;

namespace backend_asp_school.Models
{
    public class Exam
    {
        public int ExamID { get; set; }
        public string? ExamName { get; set; }
        public DateTime? ExamDate { get; set; }

        public int? SubjectID { get; set; }
        public Subject? Subject { get; set; }

        public int? ClassID { get; set; }
        public Classes? Class { get; set; }

        public int? TeacherID { get; set; }
        public Teacher? Teacher { get; set; }
    }
}
